packages/client/src/client.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  BaseError,
+  RawContractError,
+  Hex,
+  Address,
+  Abi,
+  Hash,
+  createPublicClient,
+  encodeFunctionData,
+  PrivateKeyAccount,
+  walletActions,
+  http,
+} from 'viem'
+import * as chains from 'viem/chains'
+
+import {
+  DomainData,
+  MessageData,
+  TypedSignature,
+} from '@blockful/gateway/src/types'
+import { abi as dbABI } from '@blockful/contracts/out/DatabaseResolver.sol/DatabaseResolver.json'
+
+export function getRevertErrorData(err: unknown) {
+  if (!(err instanceof BaseError)) return undefined
+  const error = err.walk() as RawContractError
+  return error?.data as { errorName: string; args: unknown[] }
+}
+
+export type CcipRequestParameters = {
+  body: { data: Hex; signature: TypedSignature; sender: Address }
+  url: string
+}
+
+export async function ccipRequest({
+  body,
+  url,
+}: CcipRequestParameters): Promise<Response> {
+  return await fetch(url.replace('/{sender}/{data}.json', ''), {
+    body: JSON.stringify(body, (_, value) =>
+      typeof value === 'bigint' ? value.toString() : value,
+    ),
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  })
+}
+
+export async function handleL2Storage({
+  chainId,
+  l2Url,
+  args,
+}: {
+  chainId: bigint
+  l2Url: string
+  args: {
+    abi: Abi | unknown[]
```

Comment:
why does it needs to receive "unknown"? Some kind of patter that we are following?

Reply:
It is a pattern used by Viem: https://github.com/wevm/viem/blob/main/src/clients/decorators/public.ts#L324